### DISCUSSION TITLE
Restrict the firebase access to just the code-server

### DISCUSSION THREAD

**Discussion by lennihein** [coder/code-server#2717] [Q&A] [answered]
Hi there,

is it possible to restrict the access of the realtime database in firebase to just the code-server?

currently my access rules are as follows:

```
{
  "rules": {
    ".read": true,
    ".write":true
  }
}
```

This leaves the database open for read and write from outside. I had no luck so far finding a possible solution to restricting the access, without also locking out code-server.

**Comment by nooobcoder** [comment, marked as answer]
@lennihein this is not supported yet, firebase is continuously evolving. I hope this feature comes out soon. You can however apply rules to modify database only if the user is authenticated kinda. 
P.S: **This is not possible as of now**

Suggestion: Use **firestore** instead of realtime databse, it is a blaze. If you are familiar with nosql databases like MongoDB then firestore should seems very similar, it has collection->document structure.

Thats all of now,
@lennihein 
Please mark the thread as answered if you are satisfied with the answer. Feel free to ask any questions.
FYI, code-server is just an app that runs as a container, **think it as vscode running on your own pc. You can set a rule to allow access by your public ip only if Google introduces such rules.**

Hope you have a great day,
Love from India 🇮🇳
Bye

**Reply by lennihein** [reply]
Aight. That's what I wanted to know, if Firebase/vscode-live were just lacking the feats or if I was stupid :)
I'll check out firestore, also.
Have a nice day, stay safe
